include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,124 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#define RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#include <type_traits>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        namespace view
+        {
+            /// \addtogroup group-views
+            /// @{
+
+            template <typename T>
+            struct linear_distribute_view
+              : view_facade<linear_distribute_view<T>, finite>
+            {
+                CONCEPT_ASSERT(std::is_arithmetic<T>());
+            private:
+                friend range_access;
+
+                T from_, to_;
+                std::ptrdiff_t n_;
+                
+                constexpr T read() const noexcept
+                {
+                    return from_;
+                }
+                constexpr bool equal(default_sentinel) const noexcept
+                {
+                    return n_ == 0;
+                }
+                CONCEPT_REQUIRES(std::is_integral<T>())
+                RANGES_CXX14_CONSTEXPR
+                bool equal(linear_distribute_view const& other) const noexcept
+                {
+                    RANGES_EXPECT(from_ == other.from_ && to_ == other.to_);
+                    return n_ == other.n_;
+                }
+                CONCEPT_REQUIRES(std::is_floating_point<T>())
+                constexpr bool equal(linear_distribute_view const& other) const noexcept
+                {
+                    return n_ == other.n_;
+                }
```

Comment:
Agreed. Copies of floating point values should be equality comparable, and floating point operations are deterministic and equality preserving to the best of my knowledge. (I hate to think of the implications if they weren't.)

I think one `equal` that does `RANGES_EXPECT(from_ == other.from_ && to_ == other.to_)` with a warning suppression would be best.